### PR TITLE
feat: ✨ Adds addOnBlur behaviour (without breaking changes)

### PR DESCRIPTION
**Pull request by MMMikeM**
I went for a discriminated union here to prevent both addOnBlur and clearOnBlur can't both be true. Let me know if you'd prefer a different implementation

### CODE REVIEW

File: src/components/TextFieldChip/TextFieldChips.tsx [hunk 1]
```diff
@@ -165,7 +142,7 @@ const TextFieldChips = React.forwardRef(
     const updateChip = (
       chipValue: MuiChipsInputChip,
       chipIndex: number,
-      event: React.KeyboardEvent<HTMLInputElement>
+      event?: React.KeyboardEvent<HTMLInputElement>
```
Comment (viclafouch) [inline src/components/TextFieldChip/TextFieldChips.tsx:145]: why it became optional ?

Comment (MMMikeM) [reply src/components/TextFieldChip/TextFieldChips.tsx:145]: handleClickAway doesn't have an event - line 183-185 I am updating the chips from within that function

Is there a different way you'd prefer for it to be handled?

Comment (viclafouch) [reply src/components/TextFieldChip/TextFieldChips.tsx:145]: Ahhh ok got it ! No it's good ;) ty !

**Comment by MMMikeM** [comment]
@viclafouch is there anything else I can do to help get this merged?

File: src/components/TextFieldChip/TextFieldChips.test.tsx [hunk 2]
```diff
@@ -138,6 +138,23 @@ describe('components/TextFieldChips', () => {
...
+        addOnBlur
+        onInputChange={callbackOnInputChange}
+      />
+    )
+
+    expect(testUtils.getInputElement().value).toBe('')
+    await testUtils.typeInInputElement('foo')
+    fireEvent.click(document)
+    expect(testUtils.getInputElement().value).toBe('')
+    expect(callbackOnInputChange).toHaveBeenCalledTimes(4)
```
Comment (viclafouch) [inline src/components/TextFieldChip/TextFieldChips.test.tsx]: add toHaveBeenCalledWith plz

Comment (MMMikeM) [reply src/components/TextFieldChip/TextFieldChips.test.tsx]: Updated - feel a lot better about that one!

Comment (MMMikeM) [reply src/components/TextFieldChip/TextFieldChips.test.tsx]: Happy with this or anything else you'd like me to add?

Comment (viclafouch) [reply src/components/TextFieldChip/TextFieldChips.test.tsx]: Let me release that this weekend ;) 
Sorry for the delay !

File: src/components/TextFieldChip/TextFieldChips.tsx [hunk 3]
```diff
@@ -190,6 +167,51 @@ const TextFieldChips = React.forwardRef(
...
+        const inputValueTrimed = inputValue.trim()
+        if (inputValue.length > 0) {
+          if (inputValueTrimed.length === 0) {
+            clearInputValue()
+          } else if (chipIndexEditable !== null) {
+            updateChip(inputValueTrimed, chipIndexEditable)
+          } else {
+            addChip(inputValueTrimed)
+          }
+        } else if (
```
Comment (viclafouch) [inline src/components/TextFieldChip/TextFieldChips.tsx]: I don't get it, you remove chip on Blur ? why ?

File: src/components/TextFieldChip/TextFieldChips.tsx [hunk 4]
```diff
@@ -190,6 +167,51 @@ const TextFieldChips = React.forwardRef(
...
+    const handleClickAway = () => {
+      if (!isFocusingRef.current) {
+        return
+      }
+      if (chipIndexEditable !== null) {
+        clearChipIndexEditable()
+        clearInputValue()
+      } else if (addOnBlur) {
+        const inputValueTrimed = inputValue.trim()
+        if (inputValue.length > 0) {
```
Comment (viclafouch) [inline src/components/TextFieldChip/TextFieldChips.tsx:178]: ```inputValueTrimed.length > 0``` ?

Comment (viclafouch) [reply src/components/TextFieldChip/TextFieldChips.tsx:179]: Ok you copied the code from `handleKeyDown`, but the feature here is only adding a chip, or update one on blur, that's all.

File: src/components/TextFieldChip/TextFieldChips.tsx (hunk 3 above)
Comment (viclafouch) [reply src/components/TextFieldChip/TextFieldChips.tsx]: Ok you copied the code from handleKeyDown, but the feature here is only adding a chip, or update one on blur, that's all.

File: src/components/TextFieldChip/TextFieldChips.test.tsx [hunk 5]
```diff
@@ -138,6 +138,18 @@ describe('components/TextFieldChips', () => {
     expect(callbackOnInputChange).toHaveBeenCalledWith('')
   })
 
+  test('should add chip on blur when addOnBlur', async () => {
```
Comment (viclafouch) [inline src/components/TextFieldChip/TextFieldChips.test.tsx:141]: would be nice to add a test for updating a chip with blur

Comment (viclafouch) [reply src/components/TextFieldChip/TextFieldChips.test.tsx:141]: And also, a test for `clearInputValue()`.

Conclusion : you should test all your `if` condition of your `handleClickAway` function

File: src/components/TextFieldChip/TextFieldChips.tsx (hunk 4 above)
Comment (MMMikeM) [reply src/components/TextFieldChip/TextFieldChips.tsx:178]: If the user clicks away while the value is " ", this clears the input field.

Would you prefer any spaces to be left in place?

File: src/components/TextFieldChip/TextFieldChips.tsx (hunk 3 above)
Comment (MMMikeM) [reply src/components/TextFieldChip/TextFieldChips.tsx]: Removed that block 👍

File: src/components/TextFieldChip/TextFieldChips.test.tsx (hunk 5 above)
Comment (MMMikeM) [reply src/components/TextFieldChip/TextFieldChips.test.tsx:141]: Thanks, will tackle this Monday morning when I'm at my IDE 👍

**Comment by Snowblaze** [comment]
Hey guys, any news? Would love to see this released

**Pull request merged**